perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -739,6 +739,23 @@ public MemberGroupStatus getUnifiedMemberGroupStatus(PerunSession sess, User use
 			namedParams.addValue("statuses", statusCodes);
 		}
 
+		String groupQueryJoin = "";
+		String groupQueryWhere = "";
+		String groupStatusesQueryString = "";
+		if (query.getGroup() != null) {
+			groupQueryJoin = " LEFT JOIN groups_members ON members.id=groups_members.member_id ";
```

Comment:
I think you don't want a `LEFT` join here. You don't want to keep records for members without groups. However, if you will rewrite the SQL, this commend might be irrelevant.